src/compiler/utilities.ts

```diff
@@ -4235,6 +4251,65 @@ namespace ts {
         return options.watch && options.hasOwnProperty("watch");
     }
 
+    /**
+     * Given a name and a list of names are *not* equal to the name, return a spelling suggestion if there is one that is close enough.
+     * Names less than length 3 only check for case-insensitive equality, not Levenshtein distance.
+     */
+    export function getSpellingSuggestion(name: string, choices: string[]): string | undefined;
+
+    /**
+     * Given a name and a list of names are *not* equal to the name, return a spelling suggestion if there is one that is close enough.
+     * Names less than length 3 only check for case-insensitive equality, not Levenshtein distance.
+     */
+    export function getSpellingSuggestion<T>(name: string, choices: T[], exclusions: string[] | undefined, getName: (candidate: T) => string | undefined): T | undefined;
+    export function getSpellingSuggestion<T>(name: string, choices: T[], exclusions?: string[], getName: (candidate: T) => string | undefined = identity): T | undefined {
+        // If there is a candidate that's the same except for case, return that.

```

Comment:
Ouch. I didn't realise that both overloads had separate jsdoc. And `@summary` is exactly what I was thinking of. Isn't there the opposite, something like `@remarks` ? (Later: nope, I guess I am thinking of XML doc comments in C#)